cmake/ExternalMFEM.cmake

```diff
@@ -0,0 +1,252 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+#
+# Build MFEM
+#
+
+# Force build order
+if(PALACE_BUILD_EXTERNAL_DEPS)
+  set(MFEM_DEPENDENCIES hypre metis)
+  if(PALACE_WITH_MUMPS)
+    list(APPEND MFEM_DEPENDENCIES mumps)
+  endif()
+  if(PALACE_WITH_STRUMPACK)
+    list(APPEND MFEM_DEPENDENCIES strumpack)
+  endif()
+  if(PALACE_WITH_SUPERLU)
+    list(APPEND MFEM_DEPENDENCIES superlu_dist)
+  endif()
+else()
+  set(MFEM_DEPENDENCIES)
+endif()
+if(PALACE_WITH_GSLIB)
+  list(APPEND MFEM_DEPENDENCIES gslib)
+endif()
```

Comment:
Why isn't this inside of the above if/else? Is GSLIB not an external dependency?

Reply:
We always build our own GSLIB, similar to NekRS and other codes. This is the same way we handle MFEM, for example. Namely there's a patch applied to GSLIB to allow for dynamic linkage and better header installation organization.